lib/coffee/AccessTokenEncryptor.coffee

```diff
@@ -1,42 +1,61 @@
 crypto = require('crypto')
+async = require('async')
 
 ALGORITHM = 'aes-256-ctr'
 
-keyFn = (password, salt, callback)->
-	return crypto.pbkdf2(password, salt, 10000, 64, 'sha1', callback)
+keyFn = (password, salt, keyLength, callback)->
+	return crypto.pbkdf2(password, salt, 10000, keyLength, 'sha1', callback)
 
 class AccessTokenEncryptor
 
 	constructor: (settings) ->
 
 		@settings = settings
-		@cipherLabel = @settings.cipherLabel
-		throw Error("cipherLabel must not contain a colon (:)") if @cipherLabel?.match(/:/)
+		@cipherLabel = "2019.1"
 
 		@cipherPassword = @settings.cipherPasswords[@cipherLabel]
 		throw Error("cipherPassword not set") if not @cipherPassword?
 		throw Error("cipherPassword too short") if @cipherPassword.length < 16
 
 	encryptJson: (json, callback) ->
 		string = JSON.stringify(json)
-		salt = crypto.randomBytes(16)
-		keyFn @cipherPassword, salt, (err, key) =>
+		async.parallel [
+			(cb) -> crypto.randomBytes(16, cb)
+			(cb) -> crypto.randomBytes(16, cb)
+		], (err, results) =>
```

Comment:
I think you may as well skip the `async.parallel` here and do `crypto.randomBytes(32, function (err, buff) { salt = buf.slice(0,8) iv = buf.slice(8,16) ...}`. It's a nice improvement to use the asynchronous crypto function though.